arch/arm/core/cortex_m/Kconfig

```diff
@@ -106,16 +106,31 @@ config ISA_THUMB2
 config CPU_CORTEX_M_HAS_BASEPRI
 	bool
 	# Omit prompt to signify "hidden" option
+	depends on ARMV7_M_ARMV8_M_MAINLINE
```

Comment:
This is a bit circular, in that we have ARMV7_M_ARMV8_M_MAINLINE select this and it depends on ARMV7_M_ARMV8_M_MAINLINE.  I think its best to remove the depends on.